### PR TITLE
Add support for WeCom (a.k.a WxWork or WeWork)

### PR DESCRIPTION
**Pull request by davinwang**
## Description
Added provider support for WeCom (a.k.a WxWork or WeWork)

## Motivation and Context
WeCom is a popular collaborate software in China for corporations. It has several oauth2 scenario, e.g. for employee, for customer, for 3rd part hosted app, QR code, etc. This PR covers the employee oauth scenario as a start. I hope other contributors will enhance it to support more scenarios.

## How Has This Been Tested?
I tested within my company setting, it should has no affect with other provider. 

## Checklist:
- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [ ] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: providers/wecom.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+)
+
+// WeComProvider represents an WeCom based Identity Provider
+type WeComProvider struct {
+	*ProviderData
+	CorpId        string
+	CorpAccessToken           string
```
Comment (JoelSpeed) [inline providers/wecom.go:19]: Why is the token at the corp level and not user level? That's a bit concerning

File: providers/wecom.go [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Acquire CorpAccessToken before access oauth2 api
+func (p *WeComProvider) getCorpAccessToken(ctx context.Context) (string, error) {
+	// return directly if valid
+	if p.CorpAccessToken != "" && p.CorpAccessTokenExpiration.After(time.Now())  {
```
Comment (JoelSpeed) [inline providers/wecom.go:180]: Mutating the provider looks ripe for a race condition 🤔

**Review by JoelSpeed — COMMENTED** [review]
I am reluctant to accept new providers at the moment for several reasons:
* We are planning a massive rewrite of how the providers are actually implemented
* We have a large number of providers for lesser known sites that have been abandoned by the implementors
* Maintainers cannot maintain implementations without access to the systems, which we don't have for most cases

Given the scope of this project already, adding new providers wouldn't be sensible right now.

Once we have the new provider implementations in place and we get things a little more sorted, I may revisit this decision, but for now, please maintain this on your own fork

**Comment by davinwang** [comment]
ok. I found adding a new provider coupling with several files, will be glad to know if the rewrite is done. So I will keep it in my own fork for now.

**Pull request closed without merging**